src/connection/ConnectionOptionsReader.ts

```diff
@@ -172,11 +176,10 @@ export class ConnectionOptionsReader {
                 Object.assign(connectionOptions, { migrations: migrations });
             }
 
-            // make database path file in sqlite relative to package.json
+            // make database path file absolute in sqlite
             if (options.type === "sqlite" || options.type === "better-sqlite3") {
                 if (typeof options.database === "string" &&
-                    options.database.substr(0, 1) !== "/" &&  // unix absolute
-                    options.database.substr(1, 2) !== ":\\" && // windows absolute
+                    !isAbsolute(options.database) &&
```

Comment:
The attaching should probable be defined within the connection options for sqlite instead of in random entities so they are in the same place as the main file's database.  The entities could then reference the alias instead of the filename.